File: problem-1/main.rs
use std::env;
use std::fs::File;
use std::io::Read;

fn sub(a: i32, b: i32) -> i32 {
    a - b
}

fn main() {
    let mut input = String::new();
    let mut file = File::open("./problem-1/INPUT").expect("Failed to open file");
    file.read_to_string(&mut input).expect("Failed to read file");

    let mut values = input.trim().split_whitespace();
    let a: i32 = values.next().expect("Missing value").parse().expect("Invalid value");
    let b: i32 = values.next().expect("Missing value").parse().expect("Invalid value");
    println!("{}", sub(a, b));
}

File: problem-1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func sub(a, b int) int {
	return a - b
}

func main() {
	file, err := os.Open("./problem-1/INPUT")
	if err != nil {
		fmt.Println("Failed to open file:", err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanWords)

	var a, b int
	if scanner.Scan() {
		a, err = strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println("Invalid argument:", scanner.Text())
			return
		}
	} else {
		fmt.Println("Failed to read argument a from file")
		return
	}

	if scanner.Scan() {
		b, err = strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println("Invalid argument:", scanner.Text())
			return
		}
	} else {
		fmt.Println("Failed to read argument b from file")
		return
	}

	result := sub(a, b)

	fmt.Println(result)
}